google/resource_compute_instance.go

```diff
@@ -1371,6 +1389,12 @@ func expandAttachedDisk(diskConfig map[string]interface{}, d *schema.ResourceDat
 			RawKey: v.(string),
 		}
 	}
+
+	if v, ok := diskConfig["kms_key_self_link"]; ok {
+		disk.DiskEncryptionKey = &computeBeta.CustomerEncryptionKey{
+			KmsKeyName: v.(string),
```

Comment:
Is it possible for `KmsKeyName` and `RawKey` to both be set at the same time? (my guess would be no, but I haven't checked.) If so, then we'll want to add that functionality to this section of code. If not, then let's explicitly document that (normally we'd also check it plan-time with `ConflictsWith`, but I don't think those work on inner elements of objects in lists).

Reply:
Only one can be set.

Should we check explicitly with something like the code below, or should we wait for the API to catch it?

```go
if v, ok := diskConfig["disk_encryption_key_raw"]; ok {
	disk.DiskEncryptionKey = &computeBeta.CustomerEncryptionKey{
		RawKey: v.(string),
	}
}

if v, kmsOk := diskConfig["kms_key_self_link"]; kmsOk {
	if ok {
		return nil, errors.New("Only one of kms_key_self_link and disk_encryption_key_raw can be set")
	}
	disk.DiskEncryptionKey = &computeBeta.CustomerEncryptionKey{
		KmsKeyName: v.(string),
	}
}
```
In terms of documentation, is documentation on the website for these values sufficient?

Reply:
My general advice is that if the API error message is good, then let it fall through to the API, otherwise add our own.

And yes, thank you!

Reply:
After a bit more investigation the API accepts both, but ignores the RawKey and uses the KMS key if both are specified. That being said, I don't think it makes sense to allow both to be specified.

Reply:
Cool, I agree with you there. Thanks for checking!